pkg/querier/querier.go

```diff
@@ -161,9 +161,7 @@ func (q *SingleTenantQuerier) SelectLogs(ctx context.Context, params logql.Selec
 
 		iters = append(iters, storeIter)
 	}
-	if len(iters) == 1 {
```

Comment:
Why is this no longer valid?

Reply:
`MergedEntryIter` is the only way we dedupe entries. Now that duplicates are being considered within a single iterator, it needs to be enforced that the returned iter is a `MergedEntryIterator`

To illustrate: [this test](https://github.com/grafana/loki/blob/25cc9402b7a076be2d3d33a9277233f233d376b0/pkg/querier/querier_test.go#L1115-L1152) will fail when that code is left in place

Reply:
I think we should keep this code as is and instead the iterators coming from the ingester|store themselves should require that they dedupe their own contents. The test you mentioned is introduced in this PR and would fail because it's introducing duplicates coming from a single store iterator. Instead, I believe it should test deduping across iterators.